packages/core/src/test/credentials/sso/server.test.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import {
+    AuthError,
+    AuthSSOServer,
+    InvalidStateError,
+    MissingCodeError,
+    MissingStateError,
+} from '../../../auth/sso/server'
+import request, { RequestError } from '../../../common/request'
+import { URLSearchParams } from 'url'
+
+describe('AuthSSOServer', function () {
+    const code = 'zfhgaiufgsbdfigsdfg'
+    const state = 'state'
+    const error = 'foo'
+    const errorDescription = 'foo'
+
+    let server: AuthSSOServer
+
+    beforeEach(async function () {
+        server = new AuthSSOServer(state, 'vscode://foo')
+        await server.start()
+    })
+
+    afterEach(async function () {
+        await server.close()
+    })
+
+    function createURL(baseUrl: string, params: Record<string, string>) {
+        const url = new URL(baseUrl)
+        url.search = new URLSearchParams(params).toString()
+        return url.toString()
+    }
+
+    async function createRequest(params: Record<string, string>, expectedErrorMsg?: string) {
```

Comment:
Should we rename this to something like `assertRequestError()`. Also does it make sense for `expectedErrorMsg` to be required?